k8up/Makefile

```diff
@@ -0,0 +1,20 @@
+MAKEFLAGS += --warn-undefined-variables
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c
+.DEFAULT_GOAL := help
+.DELETE_ON_ERROR:
+.SUFFIXES:
+
+# Git-treeish of the K8up archive version to fetch
+K8UP_VERSION     := master
+K8UP_CRD_VERSION := v1
+
+.PHONY: update_crds
+update_crds: ## Update the CRDs by fetching them from the k8up repository.
+	rm -rf crds
+	mkdir crds
+	curl -sL https://github.com/vshn/k8up/archive/$(K8UP_VERSION).tar.gz | tar -xj --directory crds --strip-components=5 k8up-master/config/crd/apiextensions.k8s.io/$(K8UP_CRD_VERSION)/
```

Comment:
FYI: on the next K8up release, there will be the all-in-one-YAML-file CRDs attached to the release itself (see example: https://github.com/vshn/espejo/releases/tag/v0.3.2), there will be no need to do git/tar/kustomize magic here.

The question is though, how we bundle two different versions of K8up CRDs. One is for K8s >= 1.16, the other for < 1.16 (e.g. Openshift 3.11). Either Helm detects that automatically or maybe we can/have to make use of `Capabilities.KubeVersion.Minor` (see https://helm.sh/docs/chart_template_guide/builtin_objects/) in an if-else construct.

Reply:
Ah, that is a nice one. As of the versioning issue: helms `crd` directory does not support templating. So I assume that `Capbabilities` is of no use here. Looking at https://helm.sh/docs/chart_best_practices/custom_resource_definitions/, we might need to go with having a separate Helm chart for the CRDs. However, I rather would like to omit them and let people install them manually or with their config management tool of choice (Project Syn in our case).

Reply:
CRDs are now available at https://github.com/vshn/k8up/releases/download/v1.0.0-rc1/k8up-crd.yaml and https://github.com/vshn/k8up/releases/download/v1.0.0-rc1/k8up-crd-legacy.yaml.